diff --git a/apple/internal/aspects/app_intents_aspect.bzl b/apple/internal/aspects/app_intents_aspect.bzl
index 4df9f0e5..81d8dd6f 100644
--- a/apple/internal/aspects/app_intents_aspect.bzl
+++ b/apple/internal/aspects/app_intents_aspect.bzl
@@ -14,6 +14,7 @@
 
 """Implementation of the aspect that propagates AppIntentsInfo providers."""
 
+load("@apple_support//lib:xcode_support.bzl", "xcode_support")
 load(
     "@apple_support//lib:apple_support.bzl",
     "apple_support",
@@ -53,7 +54,7 @@ def _app_intents_aspect_impl(target, ctx):
     if not module_names:
         module_names = [derive_swift_module_name(ctx.label)]
     xcode_version_config = ctx.attr._xcode_config[apple_common.XcodeVersionConfig]
-    if xcode_version_config.xcode_version() >= apple_common.dotted_version("15.0"):
+    if xcode_support.is_xcode_at_least_version(xcode_version_config, "15.0"):
         swiftconstvalues_files = target[OutputGroupInfo]["const_values"].to_list()
 
     return [
diff --git a/apple/internal/resource_actions/actool.bzl b/apple/internal/resource_actions/actool.bzl
index ecdbd1e1..a7ae4ee1 100644
--- a/apple/internal/resource_actions/actool.bzl
+++ b/apple/internal/resource_actions/actool.bzl
@@ -14,6 +14,7 @@
 
 """ACTool related actions."""
 
+load("@apple_support//lib:xcode_support.bzl", "xcode_support")
 load(
     "@apple_support//lib:apple_support.bzl",
     "apple_support",
@@ -189,7 +190,7 @@ Found the following legacy .appiconset files: {xcasset_appicon_files}
     )
 
     if icon_bundle_files:
-        is_xcode_26_or_later = xcode_config.xcode_version() >= apple_common.dotted_version("26.0")
+        is_xcode_26_or_later = xcode_support.is_xcode_at_least_version(xcode_config, "26.0")
         if not is_xcode_26_or_later:
             fail("""
             Found Icon Composer .icon bundles among the assigned app_icons. These are only \
